src/target/target_flash.c

```diff
@@ -312,3 +312,65 @@ bool target_flash_complete(target_s *target)
 	target_exit_flash_mode(target);
 	return result;
 }
+
+static bool flash_blank_check(target_flash_s *flash, target_addr_t src, size_t len, target_addr_t *mismatch)
+{
+	bool result = true; /* Catch false returns with &= */
+	target_s *target = flash->t;
+	platform_timeout_s timeout;
+	platform_timeout_set(&timeout, 500);
+
+	while (len) {
+		const size_t offset = src % flash->writebufsize;
+		const size_t local_len = MIN(flash->writebufsize - offset, len);
+		/* Fetch chunk into sector buffer */
+		target_mem32_read(target, flash->buf, src, local_len);
+
+		/* Compare bytewise with erased value */
+		const uint8_t erased = flash->erased;
+		for (size_t i = 0; i < local_len; i++) {
+			if (flash->buf[i] != erased) {
+				*mismatch = src + i;
+				return false;
+			}
+		}
+		src += local_len;
+		len -= local_len;
+		target_print_progress(&timeout);
+	}
+	return result;
+}
+
+bool target_flash_blank_check(target_s *target)
+{
+	if (!target->flash)
+		return false;
+
+	bool result = true;
+	target_addr_t mismatch = 0;
+
+	for (target_flash_s *flash = target->flash; flash; flash = flash->next) {
+		if (!flash->buf)
+			if (!flash_buffer_alloc(flash))
```

Comment:
`if (!flash->buf && !flash_buffer_alloc(flash))`? Reduces the nesting level and more clearly describes the condition you're aiming for here.